pkg/defaults/defaults.go

```diff
@@ -130,8 +130,11 @@ func processOpSrc(client wrapper.Client, defaultOpsrc v1.OperatorSource) error {
 		return err
 	}
 
-	// Create if not present or is deleted
-	if errors.IsNotFound(err) || !opsrcCluster.ObjectMeta.DeletionTimestamp.IsZero() {
+	// Create if not present OR the object exists AND it was cleaned up already
+	// (marked for deletion, finalizer removed) but the garbage collector hasn't gotten
```

Comment:
Nit: I suggest making it two sentences for easier understanding.  `..., finalizer removed). However if it was marked for deletion and the garbage collected hasn't gotten to it yet, i.e. the finalizers have not been removed, do nothing here and allow the deletion reconciler to remove the finalizer and recreate the object.`